code/game/atoms.dm

```diff
@@ -642,15 +642,25 @@ var/list/blood_splatter_icons = list()
 	update_icons()	//apply the now updated overlays to the mob
 
 
-/atom/proc/add_vomit_floor(mob/living/carbon/M as mob, var/toxvomit = 0)
-	if( istype(src, /turf/simulated) )
-		var/obj/effect/decal/cleanable/vomit/this = new /obj/effect/decal/cleanable/vomit(src)
+/atom/proc/add_vomit_floor(toxvomit = 0, green = FALSE)
+	playsound(src, 'sound/effects/splat.ogg', 50, 1)
+	if(!isspaceturf(src))
+		var/type = green ? /obj/effect/decal/cleanable/vomit/green : /obj/effect/decal/cleanable/vomit
+		var/vomit_reagent = green ? "green_vomit" : "vomit"
+		for(var/obj/effect/decal/cleanable/vomit/V in get_turf(src))
+			if(V.type == type)
+				if(V.scoop_reagents[vomit_reagent])
+					V.scoop_reagents[vomit_reagent] += 5
+				else
+					V.scoop_reagents[vomit_reagent] = 5

```

Comment:
Uhhhhhhh, what are you doing?

Reply:
Could probably just +=5 it either way without bothering with the if/else, huh?

Reply:
I was not sure if byond would figure it out on it's own.
If an empty spot in the dictionary would be filled correctly by just doing +=5 since you're referencing to a null pointer with that.
so space =  null + 5